pkg/reconciler/route/reconcile_resources.go

```diff
@@ -189,14 +188,14 @@ func (c *Reconciler) reconcilePlaceholderServices(ctx context.Context, route *v1
 
 		// Check if we have endpoints for this service
 		endpoints, err := c.endpointsLister.Endpoints(ns).Get(desiredService.Name)
-		if apierrs.IsNotFound(err) {
-			// noop
-		} else if err != nil {
-			return nil, err
-		} else if !metav1.IsControlledBy(endpoints, route) {
-			// Surface an error in the route's status, and return an error.
-			route.Status.MarkEndpointNotOwned(desiredService.Name)
-			return nil, fmt.Errorf("route: %q does not own Endpoints: %q", route.Name, desiredService.Name)
+		if !apierrs.IsNotFound(err) {
```

Comment:
This is not very readable. I think we have four cases:

- the error is not nil, but is an NotFound --> continue
- the error is not nil, and not a NotFound --> stop

- the error is nil, and is controlled by our route --> continue
- the error is nil, and is not controlled by our route --> stop

so maybe we can have to separate if blocks?

Reply:
I have addressed this comment.